src/go/parser/main.go

```diff
@@ -0,0 +1,55 @@
+package main
+
+import (
+	"encoding/json"
+	"go/ast"
+	"go/parser"
+	"go/token"
+	"log"
+	"os"
+)
+
+type pos struct {
+	Index  int `json:"index"`
+	Offset int `json:"offset"`
+	Line   int `json:"line"`
+	Column int `json:"column"`
+}
+
+func main() {
+	fset := token.NewFileSet()
+	file, err := parser.ParseFile(fset, os.Args[1]+".go", nil, parser.ParseComments)
```

Comment:
If you incorporate precompilation, you can remove `+".go"` here and remove `.substring(...)` in the TS